src/Stack/Config.hs

```diff
@@ -587,8 +607,37 @@ loadBuildConfig mproject maresolver mcompiler = do
             { projectResolver = fromMaybe (projectResolver project') mresolver
             }
 
-    resolver <- completeSnapshotLocation $ projectResolver project
-    (snapshot, _completed) <- loadAndCompleteSnapshot resolver
+    lockFileOutdated <- isLockFileOutdated stackYamlFP
+    if lockFileOutdated
+    then do
+      logDebug "Lock file is outdated or isn't present"
+      generatePackageLockFile stackYamlFP
```

Comment:
in `generatePackageLockFile` you complete locations when it's needed, why can't we just reuse that information and not to go through FS to get it?

Reply:
> and not to go through FS

We go through the FS to read the lock file to see if we can avoid doing `completePackageLocation` and rather reuse the information from the lock file. Is there a reason you don't want to go through FS ?

Reply:
I see no reasons not to reuse already completed locations without going through FS